### PR TITLE
cleanup: Making headers accurate on singular/plurals

### PR DESCRIPTION
**Pull request by dandhlee**
Expanding from making `Examples` into correct singular/plural forms, made a lot of other headers into its correct singular/plural forms.

Excluded the following for now, since I could not find them being used anywhere:
```
  "typesInSubtitle": "Types",
  "constsInSubtitle": "Constants",

  "namespacesInSubtitle": "Namespaces",
  "classesInSubtitle": "Classes",
  "structsInSubtitle": "Structs",
  "interfacesInSubtitle": "Interfaces",
  "enumsInSubtitle": "Enums",
  "delegatesInSubtitle": "Delegates",
  "constructorsInSubtitle": "Constructors",
  "fieldsInSubtitle": "Fields",
  "propertiesInSubtitle": "Properties",
  "methodsInSubtitle": "Methods",
  "eventsInSubtitle": "Events",
  "operatorsInSubtitle": "Operators",
  "eiisInSubtitle": "Explicit Interface Implementations",
  "functionsInSubtitle": "Functions",
  "variablesInSubtitle": "Variables",
  "typeAliasesInSubtitle": "Type Aliases",
  "membersInSubtitle": "Members",
```

### CODE REVIEW

File: third_party/docfx/templates/devsite/token.json [hunk 1]
```diff
@@ -22,20 +22,20 @@
   "improveThisDoc": "Improve this Doc",
   "viewSource": "View Source",
   "inheritance": "Inheritance",
-  "inheritedMembers": "Inherited Members",
-  "derived": "Derived Types",
+  "inheritedMember": "Inherited Member",
+  "derived": "Derived Type",
```
Comment (tbpg) [inline third_party/docfx/templates/devsite/token.json:26]: I don't think these changes are always correct. There are some classes that have multiple derived types (same with the other changes).

**Review by tbpg — CHANGES_REQUESTED** [review]
Looks like this PR has some changes from the other open PRs.

File: third_party/docfx/templates/devsite/token.json (hunk 1 above)
Comment (dandhlee) [reply third_party/docfx/templates/devsite/token.json:26]: I've added snippets to make them into plural form when necessary, for example:

```<h2>{{__global.inheritedMember}}{{#inheritedMembers.1}}s{{/inheritedMembers.1}}</h2>```

will that make it okay? Or would you rather keep them simply only in plural form?

Comment (tbpg) [reply third_party/docfx/templates/devsite/token.json:26]: I'm really just not sure about this. I'm fine to change it if that's what's best. But, it seems like having it plural is OK? "Here is the list of derived types: [single derived type]" -- that seems OK to me.

@jennyrosewood, thoughts?

Comment (dandhlee) [reply third_party/docfx/templates/devsite/token.json:26]: Jenny and I had a chat offline to discuss this and thus expanded this topic from just examples to anything else that applied. Sorry, should've noted that this was based on discussions I had with Jenny.

**Review by tbpg — COMMENTED** [review]
This has `h5` -> `h2` changes that are in other PRs. Otherwise, LGTM. Thanks!

**Comment by dandhlee** [comment]
After chatting with Jenny offline, we've agreed that the grammatical errors can be left to eliminate potential confusion. 
For the following:
```
"extensionMethod": "Extension Method",
"exception": "Exception",
"parameter": "Parameter",
"typeParameter": "Type Parameter",
"example": "Example",
"inheritedMember": "Inherited Member",
"derived": "Derived Type"
```
which ones should I revert back?

**Comment by dandhlee** [comment]
Reverted back `Exception`, `Inherited Member` and `Derived Type`.

**Review by tbpg — COMMENTED** [review]
Will wait on this until after #163 as there is some overlap (e.g. `h5` -> `h2` changes duplicated in this PR).

**Comment by dandhlee** [comment]
#163 has been merged, ready for review. PTAL!

**Review by tbpg — APPROVED** [review]

**Pull request merged**